app/src/forms/minesattestations/dataService.js

```diff
@@ -1,15 +1,34 @@
+const equal = require('fast-deep-equal');
 const log = require('npmlog');
 const { transaction } = require('objection');
 const { v4: uuidv4 } = require('uuid');
 
 const constants = require('./constants');
 const Models = require('./models');
 
+const copyAndRemoveStamps = (obj) => {
+  let items = obj;
+  if (!Array.isArray(items)) {
+    items = [obj];
+  }
+  return items.map(o => {
+    const x = {...o};
+    ['createdAt','createdBy','updatedAt','updatedBy'].forEach(p => delete x[p]);
+    return x;
+  });
+};
+
+const areTheseTheSame = (a, b) => {
+  const x = copyAndRemoveStamps(a);
+  const y = copyAndRemoveStamps(b);
+  return equal(x,y);
+};
+
 const dataService = {
 
   create: async (obj, user) => {
     if (!obj) {
-      throw Error('Industrial Camp cannot be created without data');
+      throw Error('Mines Attestation cannot be created without data');
```

Comment:
Might be worth considering "Mines Attestation" becoming a constant string to refer to. Will help with rapid refactoring in the future if we're starting the next form based on this one.